frontends/web/src/app.tsx

```diff
@@ -71,27 +72,36 @@ class App extends Component<Props, State> {
     /**
      * Gets fired when the route changes.
      */
-    private handleRoute = () => {
+    private handleRoute = async (e:RouterOnChangeArgs) => {
         if (panelStore.state.activeSidebar) {
             toggleSidebar();
         }
+
+        await sleep(0); // this must be run async
```

Comment:
why is this necessary? in my experience, sleep is usually a workaround and the real problem is elsewhere

Reply:
It's just a workaround to do an async route change, because if you do it sync preact throws an internal render error.
![Screenshot from 2021-11-15 12-57-18](https://user-images.githubusercontent.com/4956754/141813488-40129a7c-bbef-4e6c-85c1-54c56f5d635f.png)
This is the documentation for that specific version, they don't explain why handleRoute is an async function, and they wait to a promise without explaining why. I feel that the library doesn't have a straightforward way to redirect on a redirection on that version.

Reply:
But why is the sleep needed to change the function to async?

You can also remove the `async` keyword and do `return Promise.resolve()` at the end of the function, which also returns a promise without requiring the use of async/await syntax.